### DISCUSSION TITLE
looking to be able to control the submitted from user in XML toolkit

### DISCUSSION THREAD

**Discussion by BelovedIncrease** [zendtech/IbmiToolkit#177] [Q&A] [answered]
I'm using Zend 6 and using the toolkit through Websmart connections. 
When a job starts on the iSeries server, its under  QSYSWRK and QTMHHTTP as the user. 
I start from web PHP
  - it has require_once('ToolkitService.php');
  - it does user based signons , so I hav ethe IBM user associated with the web job.
  - it does a connection to the database. 
  - it retrieves the user id.   - $this->CURPRF = strtoupper(base64_decode($_SESSION['kth_login_userid']));
  - it does a Stateless TK connection: $tkConnection='';
	try
	{
		$tkConnection = ToolkitService::getInstance('*LOCAL', '', '', 'ibm_db2');
		
	}
      it does a  took kit program call. $result = $tkConnection->PgmCall('S21INMVCL', 'AULXXP3', $params, NULL, NULL);

A  CL program, set my library list, and then calls an RPG. 
RPG prints a barcode tag.  But,  I want the spool file to be under the Web user name and not QTMHHTTP. 

Does anybody know if we can control the user id associated with the TKConnection Program Call ? 

The Web PHP will be ran by many different users.  Using the User associated with the iseries connection I can better control who printed what within an output queue.

**Comment by alanseiden** [comment, marked as answer]
@BelovedIncrease Any reason you are not specifying a user and password here? They are blank.
`$tkConnection = ToolkitService::getInstance('*LOCAL', '', '', 'ibm_db2');`

**Comment by BelovedIncrease** [comment]
Thank you for pointing that out. 

$tkConnection = ToolkitService::getInstance('*LOCAL', '', '', 'ibm_db2');

I'll play with that for a while.  and see what I get . 




David Pearson
  937-663-9370
 


Confidentiality Notice:
This transmission (including any attachments) may contain confidential 
information belonging to the sender and is intended only for the use of 
the party or entity to which it is addressed. If you are not the intended 
recipient, you are hereby notified that any disclosure, copying, 
distribution, retention or the taking of action in reliance on the 
contents of this transmission is strictly prohibited. If you have received 
this transmission in error, please immediately notify the sender and erase 
all information and attachments.

**Comment by BelovedIncrease** [comment]
Thank you that was very useful and resolved the issue. 


David Pearson
  937-663-9370
 




From:   "Alan Seiden" ***@***.***>
To:     "zendtech/IbmiToolkit" ***@***.***>
Cc:     "BelovedIncrease" ***@***.***>, "Mention" 
***@***.***>
Date:   12/09/2021 01:53 AM
Subject:        Re: [zendtech/IbmiToolkit] looking to be able to control 
the submitted from user in XML toolkit (Discussion #177)



@BelovedIncrease Any reason you are not specifying a user and password 
here? They are blank.
$tkConnection = ToolkitService::getInstance('*LOCAL', '', '', 'ibm_db2');
?
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android. 



Confidentiality Notice:
This transmission (including any attachments) may contain confidential 
information belonging to the sender and is intended only for the use of 
the party or entity to which it is addressed. If you are not the intended 
recipient, you are hereby notified that any disclosure, copying, 
distribution, retention or the taking of action in reliance on the 
contents of this transmission is strictly prohibited. If you have received 
this transmission in error, please immediately notify the sender and erase 
all information and attachments.
